src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -350,27 +365,6 @@ private UnboundLambda BindAnonymousFunction(AnonymousFunctionExpressionSyntax sy
 
             var lambda = AnalyzeAnonymousFunction(syntax, diagnostics);
             var data = lambda.Data;
-            if (data.HasExplicitlyTypedParameterList)
```

Comment:
this logic moved into AnalyzeAnonymousFunction.  In particular, we want check parameters unilaterally even if there are types or no types (since we want to check modifier usage in the latter case).  so the HasExplicitlyTypedParameterList had to go.